lib/cog.ex

```diff
@@ -32,7 +32,8 @@ defmodule Cog do
      worker(Cog.TemplateCache, []),
      worker(Carrier.CredentialManager, []),
      supervisor(Cog.Relay.RelaySup, []),
-     supervisor(Cog.Command.CommandSup, [])] ++ adapter.describe_tree()
+     supervisor(Cog.Command.CommandSup, []),
+     supervisor(Module.concat(adapter, "Supervisor"), [])]
```

Comment:
I think it might be helpful to add some code to verify the supervisor module exists and log an error message if we can't find it. This won't be the case for the HipChat and Slack adapters but I could see this happening with a user-provided adapter. Something like `Code.ensure_loaded/1` should do the trick.